shared/constants/types/fs.js

```diff
@@ -2,21 +2,40 @@
 import * as I from 'immutable'
 
 export opaque type Path = ?string
-export type FolderPathType = 'folder'
-export type FilePathType = 'file' | 'symlink' | 'exec'
-export opaque type PathType = FolderPathType | FilePathType
-export type _FilePathItem = {
-  type: FilePathType,
-}
+
+export type PathType = 'folder' | 'file' | 'symlink' | 'exec' | 'unknown'
```

Comment:
I think this can be flat for now.
i added unknown as its possible you have some mounted component and the store doesn't have a value (due to some race). we shouldn't throw in this case really, likely want to just render nothing